src/server/rdb_load.cc

```diff
@@ -1753,6 +1757,15 @@ error_code RdbLoader::Load(io::Source* src) {
       continue;
     }
 
+    if (type == RDB_OPCODE_JOURNAL_BLOB) {
+      // We should flush all changes on the current db before applying incremental changes.
+      for (unsigned i = 0; i < shard_set->size(); ++i) {
+        FlushShardAsync(i);
+      }
```

Comment:
This is important. This way we make sure the journal entries executed only after their corresponding rdb has been parsed and inserted